lsp-mode.el

```diff
@@ -5646,29 +5646,32 @@ perform the request synchronously."
            (lsp-request "workspace/symbol" `(:query ,pattern))))
 
 (defun lsp--get-symbol-to-rename ()
-  "Get symbol to rename and placeholder at point."
-  (if (let ((rename-provider (or (lsp--capability :renameProvider)
-                                 (-some-> (lsp--registered-capability "textDocument/rename")
-                                   (lsp--registered-capability-options)))))
-        (lsp:rename-options-prepare-provider? rename-provider))
-      (-when-let (response (lsp-request "textDocument/prepareRename"
-                                        (lsp--text-document-position-params)))
+  "Get symbol to rename and placeholder at point.
+Returns a cons (SYMBOL-TO-RENAME . PLACEHOLDER). Returns nil if
+renaming is supported but cannot be done at point."
+  (unless (lsp-feature? "textDocument/rename")
+    (error "The connected server(s) doesn't support renaming"))
+  (if (lsp-feature? "textDocument/prepareRename")
+      (when-let ((response
+                  (lsp-request "textDocument/prepareRename"
+                               (lsp--text-document-position-params))))
         (-let* (((start . end) (lsp--range-to-region
                                 (if (lsp-range? response)
                                     response
                                   (lsp:prepare-rename-result-range response))))
-                (symbol (buffer-substring-no-properties start end))
+                (symbol (buffer-substring start end))
                 (placeholder (lsp:prepare-rename-result-placeholder response)))
           (cons symbol (or placeholder symbol))))
-    (let ((symbol (thing-at-point 'symbol t)))
-      (cons symbol symbol))))
+    (when-let ((sym (thing-at-point 'symbol)))
```

Comment:
why do you rename symbol? Why do you introduce when-let? 

Let's clarify that for future reference - do not introduce random renames, restructuring, etc - nobody is doing this PR in this repo. You are making the life of the reviewers harder than it should be. I have to go and test each that you have changed to verify that each line of all lines that you have changed does not introduce a regression. I cannot spend that much time reviewing trivial changes chasing for regressions.

Reply:
Ok, I want to do that in the future. The `when-let` is because that function should return nil if there is nothing to rename at point. Without that `when-let`, if `(thing-at-point symbol)` were to yield nil, the function would return `(cons nil nil)`, instead of nil as it should. This, unlike I had assumed, didn't cause any real problems, because -if-let checks the bindings, not the input against nil.

```emacs-lisp
(eq nil nil) => true
(eq nil (cons nil nil)) => false
(equal nil (cons nil nil)) => false
```

Reply:
You will still rename all local variables, introduce random changes without documenting/testing them based on assumptions and you will expect the rest of the team to do that for you?!

Reply:
Sorry, I missspelled won't...